lib/importer/data.rb

```diff
@@ -0,0 +1,113 @@
+require 'csv'
+
+module Importer
+  class Data
+
+    def initialize file_path, params=nil, delete_file=true
+      @file_name = file_path
+      @params = params
+      @delete_file = delete_file
+    end
+
+    def import_products
+      file = File.open(@file_name)
+      content = file.read
+      i = 0
+      CSV.parse( content) do |row|
+        # Skip first line which contains the header
+        i += 1
+        next if (i == 1)
+        # Initialize new product and new Campaign
+        product = Product.new()
+        campaign = Campaign.new()
+        product_images_url = []
+
+        get_csv_header[:file_columns].each_with_index do |csv_column, index|
+          # Find db_column based on csv_column name
+          db_column = @params[:product].find{|i| i[1] == csv_column}.try(:first)
+          product[db_column] = row[index] if db_column.present?
+
+          # If there is brand column in csv
+          brand_db_column = @params[:brand].find{|i| i[1] == csv_column}.try(:first)
+          if @params[:brand].present? && brand_db_column.present?
+            brand = Brand.find_or_create_by(name: row[index])
+            product[:brand_id] = brand.id
+          end
+
+          # If there is brand column in csv
+          widget_db_column = @params[:widget].find{|i| i[1] == csv_column}.try(:first)
+          if @params[:widget].present? && brand_db_column.present?
+            widget = Widget.find_or_create_by(name: row[index])
+            product[:widget_id] = widget.id
+          end
+
+          # If there is campaign column in csv
+          campaign_db_column = @params[:campaign].find{|i| i[1] == csv_column}.try(:first)
+          if @params[:campaign].present? && campaign_db_column.present?
+            campaign[campaign_db_column] = row[index]
+          end
+
+          if (csv_column == "image_urls")
+            product_images_url = row[index].split(",")
```

Comment:
Aici urma sa te intreb, pentru ca doar un url este in csv-uri pentru imagine...Pot fi si mai multe url-uri? si daca da? cum sunt separate?